hadoop-hdds/rocks-native/src/main/patches/rocks-native.patch

```diff
@@ -16,592 +16,557 @@
-       } else {
--        fprintf(stdout, "The file is ok\n");
-+        fprintf(out, "The file is ok\n");
-       }
-       continue;
-     }
-@@ -503,15 +514,15 @@ int SSTDumpTool::Run(int argc, char const* const* argv, Options options) {
-           table_properties_from_reader;
-       st = dumper.ReadTableProperties(&table_properties_from_reader);
-       if (!st.ok()) {
--        fprintf(stderr, "%s: %s\n", filename.c_str(), st.ToString().c_str());
--        fprintf(stderr, "Try to use initial table properties\n");
-+        fprintf(err, "%s: %s\n", filename.c_str(), st.ToString().c_str());
-+        fprintf(err, "Try to use initial table properties\n");
-         table_properties = dumper.GetInitTableProperties();
-       } else {
-         table_properties = table_properties_from_reader.get();
-       }
-       if (table_properties != nullptr) {
-         if (show_properties) {
--          fprintf(stdout,
-+          fprintf(out,
-                   "Table Properties:\n"
-                   "------------------------------\n"
-                   "  %s",
-@@ -523,18 +534,18 @@ int SSTDumpTool::Run(int argc, char const* const* argv, Options options) {
-         total_index_block_size += table_properties->index_size;
-         total_filter_block_size += table_properties->filter_size;
-         if (show_properties) {
--          fprintf(stdout,
-+          fprintf(out,
-                   "Raw user collected properties\n"
-                   "------------------------------\n");
-           for (const auto& kv : table_properties->user_collected_properties) {
-             std::string prop_name = kv.first;
-             std::string prop_val = Slice(kv.second).ToString(true);
--            fprintf(stdout, "  # %s: 0x%s\n", prop_name.c_str(),
-+            fprintf(out, "  # %s: 0x%s\n", prop_name.c_str(),
-                     prop_val.c_str());
-           }
-         }
-       } else {
--        fprintf(stderr, "Reader unexpectedly returned null properties\n");
-+        fprintf(err, "Reader unexpectedly returned null properties\n");
-       }
-     }
-   }
-@@ -555,9 +566,9 @@ int SSTDumpTool::Run(int argc, char const* const* argv, Options options) {
-     // Exit with an error state
-     if (dir) {
-       fprintf(stdout, "------------------------------\n");
--      fprintf(stderr, "No valid SST files found in %s\n", dir_or_file);
-+      fprintf(err, "No valid SST files found in %s\n", dir_or_file);
-     } else {
--      fprintf(stderr, "%s is not a valid SST file\n", dir_or_file);
-+      fprintf(err, "%s is not a valid SST file\n", dir_or_file);
-     }
-     return 1;
-   } else {
++Slice RawSstFileReaderIterator::getValue() const {
++    return iter_->value();
++}
++}  // namespace ROCKSDB_NAMESPACE
++
++#endif  // ROCKSDB_LITE
+diff --git a/tools/raw_sst_file_reader_iterator.h b/tools/raw_sst_file_reader_iterator.h
+new file mode 100644
+index 000000000..79313264f
+--- /dev/null
++++ b/tools/raw_sst_file_reader_iterator.h
```

Comment:
nit: let's just call it `raw_sst_file_iterator`. I think it is good enough to say `raw_sst_file_reader` provides a `raw_sst_file_iterator`.

Reply:
done